### PR TITLE
Update java IO dependencies

### PR DESCRIPTION
**Pull request by pawelpasterz**
Update a couple of java IO dependencies

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by pawelpasterz** [comment]
Run CommunityMetrics PreCommit

**Comment by pawelpasterz** [comment]
Run Java PreCommit

**Comment by pawelpasterz** [comment]
Run Java PostCommit

**Comment by pawelpasterz** [comment]
Run Java HadoopFormatIO Performance Test

**Comment by pawelpasterz** [comment]
Run Java ParquetIO Performance Test

**Comment by pawelpasterz** [comment]
Run Java ParquetIO Performance Test HDFS

**Comment by pawelpasterz** [comment]
Run Python_PVR_Flink PreCommit

**Comment by pawelpasterz** [comment]
Run Java PostCommit

**Comment by pawelpasterz** [comment]
Hey, @iemejia could you take a look? Thanks
R: @iemejia

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -536,8 +535,8 @@ class BeamModulePlugin implements Plugin<Project> {
         jackson_dataformat_yaml                     : "com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:$jackson_version",
         jackson_datatype_joda                       : "com.fasterxml.jackson.datatype:jackson-datatype-joda:$jackson_version",
         jackson_module_scala                        : "com.fasterxml.jackson.module:jackson-module-scala_2.11:$jackson_version",
-        jaxb_api                                    : "jakarta.xml.bind:jakarta.xml.bind-api:$jaxb_api_version",
-        jaxb_impl                                   : "com.sun.xml.bind:jaxb-impl:$jaxb_api_version",
+        jaxb_api                                    : "jakarta.xml.bind:jakarta.xml.bind-api:3.0.0-RC3",
```
Comment (iemejia) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:538]: This one seems to be Java 11 only so we cannot use it or we will break Java 8 users.

File: sdks/java/io/parquet/build.gradle [hunk 2]
```diff
@@ -25,7 +25,7 @@ applyJavaNature(
 description = "Apache Beam :: SDKs :: Java :: IO :: Parquet"
 ext.summary = "IO to read and write on Parquet storage format."
 
-def parquet_version = "1.10.0"
+def parquet_version = "1.11.1"
```
Comment (iemejia) [inline sdks/java/io/parquet/build.gradle:28]: This one you cannot do because of transitive dependency with Avro 1.9 BEAM-8388 for more details.

**Review by iemejia — CHANGES_REQUESTED** [review]
So many simultaneous updates are a bit error-prone but still good to have most of this. Please execute if possible the IT version of the tests to be sure we don't break anything.
Fix my comments and get it green and I suppose it would be good to merge.

CC @aromanenko-dev for extra comments + test suggestions (in AWS/Kinesis modules)

**Comment by pawelpasterz** [comment]
> So many simultaneous updates are a bit error-prone but still good to have most of this. Please execute if possible the IT version of the tests to be sure we don't break anything.
> Fix my comments and get it green and I suppose it would be good to merge.
> 
> CC @aromanenko-dev for extra comments + test suggestions (in AWS/Kinesis modules)

@iemejia 
Would you like me to divide this PR into smaller ones (one per each dependency)? It's not a problem for me

**Comment by pawelpasterz** [comment]
Run Java AvroIO Performance Test

**Comment by pawelpasterz** [comment]
Run Java AvroIO Performance Test HDFS

**Comment by pawelpasterz** [comment]
Run Java CompressedTextIO Performance Test

**Comment by pawelpasterz** [comment]
Run Java CompressedTextIO Performance Test HDFS

**Comment by pawelpasterz** [comment]
Run Java InfluxDbIO Performance Test

**Comment by pawelpasterz** [comment]
Run Java ManyFilesTextIO Performance Test

**Comment by pawelpasterz** [comment]
Run Java ManyFilesTextIO Performance Test HDFS

**Comment by pawelpasterz** [comment]
Run Java MongoDBIO Performance Test

**Comment by pawelpasterz** [comment]
Run Java ParquetIO Performance Test

**Comment by pawelpasterz** [comment]
Run Java ParquetIO Performance Test HDFS

**Comment by pawelpasterz** [comment]
Run Java TFRecordIO Performance Test

**Comment by pawelpasterz** [comment]
Run Java TFRecordIO Performance Test HDFS

**Comment by pawelpasterz** [comment]
Run Java XmlIO Performance Test

**Comment by pawelpasterz** [comment]
Run Java XmlIO Performance Test HDFS

**Comment by aromanenko-dev** [comment]
Thanks for working on this! I'd suggest to split it into several PRs (according to updated module) and link with corresponding Jira (so it will be easier to have in release notes). I know it will require more additional work but it will be better in terms of changes atomicity.

**Comment by iemejia** [comment]
Change LGTM but maybe is worth to do the separate PRs (sorry for the annoyance) because it also will be more explicit on the release notes if we create a JIRA per related update.

**Comment by pawelpasterz** [comment]
@aromanenko-dev @iemejia 
Totally agree with both of you. I'll create new PRs and (if it's ok) tag you for review

**Pull request closed without merging**